### ISSUE TITLE
Support custom DataFormats

### ISSUE THREAD

**Issue by lordrip** [KaotoIO/kaoto#1017]
### Please describe the feature that you want to propose

At the moment of writing, only DataFormats provided through the Camel Catalog are supported.

The goal for this ticket would be to add option to be able to select a custom DataFormat or to provide a reference to a DataFormat bean. Ideally, this should render a Bean configuration field that allows the user to pick an existing bean or to create one.

![image](https://github.com/KaotoIO/kaoto/assets/16512618/b36269dd-57bd-43d1-bb51-92824cb9b7db)

**Comment by igarashitm** [comment]
We already have `custom` dataformat in the form. The `ref` is supposed to be a `BeanReferenceField`, but it seems not ATM - it could be a bug.
![Screenshot from 2024-04-16 12-22-25](https://github.com/KaotoIO/kaoto/assets/265462/264e4382-8552-4d67-b4eb-5e00cc4ace74)

**Comment by igarashitm** [comment]
Here is the model catalog entry - https://github.com/apache/camel/blob/main/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/models/custom.json

**Comment by lhein** [comment]
yep, we should make it cross connect to the beans tab / enhancing the text field as a typeahead dropdown with a button for create

**Comment by igarashitm** [comment]
The tricky part for filtering beans is that it requires Java class hierarchy knowledge, for example we don't know `org.postgresql.ds.PGConnectionPoolDataSource` implements `javax.sql.DataSource`

We need either
- Retrieve class hierarchy data from vscode (rather vscode Java extension?)
- Have a dictionary for the well known implementation classes such as SQL DataSource, JMS ConnectionFactory, etc
